### PR TITLE
Faster iteration in TagHelpersRunner

### PR DESCRIPTION
**Pull request by benaadams**
Iterate over `Span<ITagHelper>` rather than `IList<ITagHelper>`; also use span sort rather than custom bubble sort

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
Thanks for your PR, @benaadams .
@ajaybhargavb can you please review this? Thanks!

**Comment by BrennanConroy** [comment]
@mkArtakMSFT can you assign someone else now?

File: src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs [hunk 1]
```diff
@@ -28,24 +29,23 @@ public Task RunAsync(TagHelperExecutionContext executionContext)
...
 
             var tagHelperContext = executionContext.Context;
-            var tagHelpers = executionContext.TagHelpers;
-            OrderTagHelpers(tagHelpers);
+            var tagHelpers = CollectionsMarshal.AsSpan(executionContext.TagHelperList);
 
-            // Read interface .Count once rather than per iteration
-            var count = tagHelpers.Count;
-            for (var i = 0; i < count; i++)
+            tagHelpers.Sort(default(SortTagHelpers));
```
Comment (pranavkm) [inline src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs:34]: Out of curiosity, is there a reason this is preferable to calling `executionContext.TagHelperList.Sort(default(SortTagHelpers))`?

File: src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs [hunk 2]
```diff
@@ -28,24 +29,23 @@ public Task RunAsync(TagHelperExecutionContext executionContext)
...
+
+            foreach (var tagHelper in tagHelpers)
```
Comment (pranavkm) [inline src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs:36]: Why iterate over the sequence two separate ways?

Comment (benaadams) [reply src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs:36]: Want `i` in the second one?

File: src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs (hunk 1 above)
Comment (benaadams) [reply src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs:34]: List Sort will create a Span and then use Span Sort; however we want the Span anyway; so might as well get that first and then sort the Span?

Iterating the Span is way faster than iterating the List; as it doesn't bounds check (or version check for `foreach`) each iteration etc, which is mostly what this is about.

**Review by pranavkm — APPROVED** [review]
Thanks!

**Pull request merged**